### PR TITLE
Release 4.4.1

### PR DESCRIPTION
**Pull request by KengoTODA**
This is a patch update because it adds no new feature.

### Changed
- Bump gson from 2.8.7 to 2.8.8 ([#1658](https://github.com/spotbugs/spotbugs/pull/1658))
- Lower `ExitCodes` logger to debug level ([#1661](https://github.com/spotbugs/spotbugs/issues/1661))
- Fixed SARIF format to be compatible with Github code scanning API requirements ([#1630](https://github.com/spotbugs/spotbugs/issues/1630))

### Fixed
- Fixed immutable classes in java.net.* as being flagged as EI ([#1653](https://github.com/spotbugs/spotbugs/issues/1653)
- Classes containing only static methods with setter-like names are no longer considered as mutable ([#1601](https://github.com/spotbugs/spotbugs/issues/1601))
- Handle all immutable collections in the Guava library as immutable ([#1601](https://github.com/spotbugs/spotbugs/issues/1601))
- Classes annotated with @Immutable or @jdk.internal.ValueBased are considered as immutable ([#1601](https://github.com/spotbugs/spotbugs/issues/1601))
- All classes in packages java.time and java.math are now correctly handled as immutable ([#1601](https://github.com/spotbugs/spotbugs/issues/1601))

### CODE REVIEW

**Comment by KengoTODA** [comment]
better to wait until #1677 and #1676 will be merged to the `master` branch.

**Comment by h3xstream** [comment]
@KengoTODA Can you integrate this PR https://github.com/spotbugs/spotbugs/pull/1631 in this release ?

**Comment by KengoTODA** [comment]
@h3xstream done. It will be a part of the 4.4.1 release!

**Review by ThrawnCA — APPROVED** [review]

**Pull request merged**
